compiler/dyno/lib/resolution/scope-queries.cpp

```diff
@@ -885,17 +884,18 @@ doResolveUseStmt(Context* context, const Use* use,
     const Scope* foundScope = findUseImportTarget(context, scope, r,
                                                   expr, VIS_USE, oldName);
     if (foundScope != nullptr) {
-      // First, add VisibilitySymbols entry for the symbol itself
-      if (newName.isEmpty()) {
-        r->addVisibilityClause(foundScope, VisibilitySymbols::SYMBOL_ONLY,
-                               isPrivate, convertOneName(oldName));
-      } else {
+      // First, add VisibilitySymbols entry for the symbol itself,
+      // iff the use is renamed or non-public
+      if (!newName.isEmpty()) {
         r->addVisibilityClause(foundScope, VisibilitySymbols::SYMBOL_ONLY,
                                isPrivate, convertOneRename(oldName, newName));
+      } else if (isPrivate) {
+        r->addVisibilityClause(foundScope, VisibilitySymbols::SYMBOL_ONLY,
+                               isPrivate, convertOneName(oldName));
       }
 
       // Then, add the entries for anything imported
-      VisibilitySymbols::Kind kind = VisibilitySymbols::ALL_CONTENTS;
+      VisibilitySymbols::Kind kind;
```

Comment:
In this case, `kind` is only used within branches of the switch, not after it, so to me this is logically not a 'default' value. Giving `kind` a value before the switch (which will never be used) wrongly implies it is. (I also think that since it's not used after the switch, compilers wouldn't complain, but I might be wrong.)